src/vs/workbench/parts/extensions/electron-browser/extensionEditor.ts

```diff
@@ -250,6 +254,7 @@ export class ExtensionEditor extends BaseEditor {
 		this.navbar.onChange(this.onNavbarChange.bind(this, extension), this, this.transientDisposables);
 		this.navbar.push(NavbarSection.Readme, localize('details', "Details"));
 		this.navbar.push(NavbarSection.Contributions, localize('contributions', "Contributions"));
+		this.navbar.push(NavbarSection.Changelog, localize('changelog', "Changelog"));
```

Comment:
The problem with that is that the network request could be slow and you'd show the link sometime after the fact. Not to mention that you should then hook up a cancellation request to that, lest the user clicks on another extension; you wouldn't want that request to still be running right?

The `hasChangelog()` method may appear ugly, but it's definitely a good solution.